### PR TITLE
use alt to focus the menu and and navigation

### PR DESCRIPTION
**Pull request by sbatten**
Please try this out with toggle setting enabled and without. Lots of code churn here because my previous focused behavior was poorly defined. I don't know how much better the new implementation but the idea is to work like a state machine with the following guidelines. If you plan to change `MenubarState`, update the focused menu and set the new state. If you are not modifying the state, you must handle the menu behavior yourself.

### CODE REVIEW

File: src/vs/base/browser/ui/menu/menu.ts [hunk 1]
```diff
@@ -321,7 +321,7 @@ class SubmenuActionItem extends MenuActionItem {
 	}
 
 	private createSubmenu() {
-		if (!this.parentData.submenu) {
+		if (!this.parentData.submenu && this.builder) {
```
Comment (bpasero) [inline src/vs/base/browser/ui/menu/menu.ts]: @sbatten how can this happen that we react on click but have not been created?

**Review by bpasero — APPROVED** [review]
@sbatten I like the way it works now, good job. I think I found one minor bug:
* `Alt`-click to focus menu
* navigate to `Terminal `and execute `Clear`
* `Alt`+click again

=> the menu does not have focus (but menmonics show up)

File: src/vs/base/browser/ui/menu/menu.ts (hunk 1 above)
Comment (sbatten) [reply src/vs/base/browser/ui/menu/menu.ts]: @bpasero converting to runoncescheduler fixed the issue, reverted this change

**Comment by sbatten** [comment]
@jakub-g this change affects the new custom menubar in insiders. 
To enable, change the setting `window.titleBarStyle` to `custom` in your `settings.json`

**Comment by iansan5653** [comment]
@jakub-g Make sure you have **Window: Title Bar Style** set to "custom".

**Pull request merged**